### DISCUSSION TITLE
How to rebuild frontend dist folder with some custom changes?

### DISCUSSION THREAD

**Discussion by surajsrivathsa** [getredash/redash#6651] [Q&A] [answered]
I am trying to change redash host path to a non-root url prefix. Until now i was somewhat successful in making this happen by changing files inside `./client/dist` and `redash` folders. But i need to apply some custom html templates, css and others for frontend and rebuild the dist folder. 

So far i tried `ng build` , `npm run build` but they need sudo access which `redash` user does not have. I also tried adding a user with sudo priveleges to the docker image but it is also not working. 

If possible can you provide password for default `redash` user or a way to rebuild the `dist` directory in redash v8 docker.

**Comment by justinclift** [comment]
Hmmm, v8 is pretty old and doesn't use the same development approach that our most recent source code does.  To keep you working with v8, lets try and figure out something that'll work for you...

> but they need sudo access which redash user does not have

Ahhh.  That sounds like you're entering into the docker container from the command line, and you're telling docker to use the default (for the container) user.  That'll be why it's putting you in the `redash` user.

You can alternatively tell docker to put you in the container with a different user, such as root.  For example:

```
$ docker exec -it --user root NAME_OF_CONTAINER /bin/bash
```

Maybe give that a go? :smile:

**Reply by surajsrivathsa** [reply]
Thank you, yeah i have access now, i just need to rebuild

**Comment by surajsrivathsa** [comment, marked as answer]
Process used to build custom reddish.

1. Login to docker by running `docker exec -it --user root NAME_OF_CONTAINER /bin/bash`.
2. Change the code inside redash and save the changes.
3. Change web pack config, static paths if required.
4. If your node is of version < 10, then try to install node 12 first. Similarly upgrade npm to v6.
5. Rebuild code using command `npm run build`.
6. Verify whether the old Dist folder is removed and recreated freshly
7. Close the IDE and restart just redash-server docker by running `docker restart NAME_OF_CONTAINER`.
